httpcodec/response.go

```diff
@@ -0,0 +1,59 @@
+package httpcodec
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// SerializeHTTPResponse take an http.Response object and serializes it into a byte
+// slice that can be embedded into another struct, such as RelayResponse.Payload.
+func SerializeHTTPResponse(response *http.Response) (body []byte, err error) {
+	responseBodyBz, err := io.ReadAll(response.Body)
+	response.Body.Close()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	if err != nil {
		return nil, err
	}
	response.Body.Close()	
```
![Screenshot 2024-05-31 at 2 20 17 PM](https://github.com/pokt-network/shannon-sdk/assets/1892194/f9aa5485-62f3-4ca1-b2e1-64eddbb004d1)